### PR TITLE
Fix the detach xml can not find error for watchdog

### PR DESCRIPTION
**Pull request by kylazhang**
Signed-off-by: Kyla Zhang <weizhan@redhat.com>

```
 (1/1) type_specific.io-github-autotest-libvirt.watchdog.action_shutdown.model_i6300esb.hotunplug_test: PASS (29.55 s)
```

### CODE REVIEW

File: libvirt/tests/src/virtual_device/watchdog.py [hunk 1]
```diff
@@ -243,7 +245,8 @@ def _check_dump_file(dump_path, domain_id):
         if hotunplug_test:
             cur_xml = vm_xml.VMXML.new_from_dumpxml(vm_name)
             cur_watchdog = cur_xml.xmltreefile.find('devices/watchdog')
-            cur_watchdog_xml = Watchdog.new_from_element(cur_watchdog).xml
+            cur_watchdog_xml = os.path.join(data_dir.get_data_dir(), "watchdog.xml")
+            shutil.copyfile(Watchdog.new_from_element(cur_watchdog).xml, cur_watchdog_xml)
```
Comment (dzhengfy) [inline libvirt/tests/src/virtual_device/watchdog.py:249]: When is the file deleted?

Comment (kylazhang) [reply libvirt/tests/src/virtual_device/watchdog.py:249]: Updated, thanks

**Review by dzhengfy — APPROVED** [review]
LGTM

**Comment by kylazhang** [comment]
```
(1/1) type_specific.io-github-autotest-libvirt.watchdog.action_shutdown.model_i6300esb.hotunplug_test: PASS (25.73 s)
```

**Review by Yingshun — APPROVED** [review]
LGTM

**Comment by Yingshun** [comment]
@dzhengfy code has been changed since your last review, would you like to re-review this PR?(and merge it)

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**